agent/proxycfg/testing_mesh_gateway.go

```diff
@@ -7,16 +7,16 @@ import (
 	"math"
 	"time"
 
-	"github.com/mitchellh/go-testing-interface"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-
 	"github.com/hashicorp/consul/acl"
 	"github.com/hashicorp/consul/agent/configentry"
 	"github.com/hashicorp/consul/agent/connect"
 	"github.com/hashicorp/consul/agent/consul/discoverychain"
 	"github.com/hashicorp/consul/agent/structs"
 	"github.com/hashicorp/consul/proto/private/pbpeering"
+	"github.com/mitchellh/go-testing-interface"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"k8s.io/utils/pointer"
```

Comment:
can we write/use an existing `toPointer` method rather than pull in this dep for this? could be something like

```
func pointerTo[T any] (v T) *T {
   return &v
}
```